### ISSUE TITLE
problems about training the texture model

### ISSUE THREAD

**Issue by happyday521** [Zhangjinso/PISE#25]
Hi, I have some problems about the difference between the pose transfer model and the texture model.
1、As you said, we can uncomment line 162-176 to train a texture model. In my opinion, it mainly changes the predicted par2 from Float to Int by the torch.argmax operation. What's the advantage or motivation of you doing it rather than use the predicted par2 directly?
2、Since argmax operation is non-differentiable, if we uncomment line 162-176 to train a texture model, the image generator can not provide the gradients for the parsing generator. Thus, the pre-trained parsing generator will not be updated during the training.  Will it affect the quality of final generated images? 
Besides, since the parsing generator are disconnected from the image generator, need we to calculate the parsing loss like loss_par and loss_par1 when tarining the image generator?

Thanks!

**Comment by Zhangjinso** [comment]
1. For parsing editing, we can only edit the 'int parsing'. If we train the image generator with 'float parsing', the results may be strange. For texture transfer, I think it is ok to use the predicted par2 directly. 
2. If use argmax, the parsing generator gets gradients from parsing loss. If you only train the image generator with the pre-trained parsing generator, the parsing loss is of no use. But recommend to train paring generator rather than use a pre-trained checkpoint.

**Comment by happyday521** [comment]
Got it. Thanks very much!
